src/modules/job-ingest/validate.c

```diff
@@ -165,28 +160,37 @@ struct validate *validate_create (flux_t *h,
 
     argc = argz_count (argz, argz_len);
     if (!(argv = calloc (1, sizeof (char *) * (argc + 1)))) {
-        flux_log_error (h, "failed to create argv");
+        flux_log_error (v->h, "failed to create argv");
         goto error;
     }
     argz_extract (argz, argz_len, argv);
 
-    for (i = 0; i < MAX_WORKER_COUNT; i++) {
-        char name[256];
-        (void) snprintf (name, sizeof (name), "validator[%d]", i);
-        if (!(v->worker[i] = worker_create (h,
-                                            worker_inactivity_timeout,
-                                            name,
-                                            argc, argv)))
+    for (int i = 0; i < MAX_WORKER_COUNT; i++) {
+        if (!v->worker[i]) {
+            char name[256];
+            (void) snprintf (name, sizeof (name), "validator[%d]", i);
+            if (!(v->worker[i] = worker_create (v->h,
+                                                worker_inactivity_timeout,
+                                                name)))
+                goto error;
+        }
+        if (worker_set_cmdline (v->worker[i], argc, argv) < 0)
             goto error;
     }
-    free (argv);
-    free (argz);
-    return v;
+    rc = 0;
 error:
     free (argv);
     free (argz);
```

Comment:
need to save/restore errno here (since errors from this function are reported with `flux_log_error()` which expects errno to be set)